lib/fluent/logger/fluent_logger.rb

```diff
@@ -230,8 +230,12 @@ def write(msg)
           end
 
           begin
-            send_data(@pending)
-            @pending = nil
+            len = send_data(@pending)
+            if len != pending_bytesize
+              @pending = @pending.slice(len, pending_bytesize)
```

Comment:
If slice the pending data, next `send_data` sends broken data, right?

Reply:
Good point… this implementation doesn't work 😢 
It is necessary to send msgpacked data all at once in one call of Fluent::Logger#post.

Reply:
fixed. https://github.com/fluent/fluent-logger-ruby/pull/80/commits/8d53f600c96062faf503068d4f533775b1f0601a